File: api-escola-conectada/src/main/java/br/edu/escolaconectada/model/Professor.java
package br.edu.escolaconectada.model;
import jakarta.persistence.*;
@Entity
public class Professor {
 @Id @GeneratedValue(strategy = GenerationType.IDENTITY) private Long id;
 private String nome; private String email; private String disciplina;
 public Long getId(){return id;} public void setId(Long id){this.id=id;}
 public String getNome(){return nome;} public void setNome(String nome){this.nome=nome;}
 public String getEmail(){return email;} public void setEmail(String email){this.email=email;}
 public String getDisciplina(){return disciplina;} public void setDisciplina(String disciplina){this.disciplina=disciplina;}
}

File: app_escola_conectadaon/lib/models/professor.dart
class Professor {
  int? id;
  String nome;
  String email;
  String disciplina;
  Professor(
      {this.id,
      required this.nome,
      required this.email,
      required this.disciplina});
  Map<String, dynamic> toJson() =>
      {'id': id, 'nome': nome, 'email': email, 'disciplina': disciplina};
  factory Professor.fromJson(Map<String, dynamic> json) => Professor(
      id: json['id'],
      nome: json['nome'] ?? '',
      email: json['email'] ?? '',
      disciplina: json['disciplina'] ?? '');
}
